CRM/Batch/Form/Entry.php

```diff
@@ -280,6 +280,11 @@ public static function formRule($params, $files, $self) {
 
     //CRM-16480 if contact is selected, validate financial type and amount field.
     foreach ($params['field'] as $key => $value) {
+      if (isset($value['trxn_id'])) {
+        if (0 < CRM_Core_DAO::singleValueQuery('SELECT id FROM civicrm_contribution WHERE trxn_id = %1', array(1 => array($value['trxn_id'], 'String')))) {
```

Comment:
I think it comparison should be made against NULL as trxn_id can be alphanumeric